In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
pd.set_option('display.max_columns', None)

In [3]:
data = pd.read_csv(f"../labeled_data/1.csv")

In [4]:
print("Shape", data.shape)
data.describe()

Shape (558714, 32)


PacketCounter  SampleTimeFine          Acc_X          Acc_Y  \
count  558714.000000             0.0  558656.000000  558656.000000   
mean    29984.349316             NaN       7.265798       0.567765   
std     17730.810665             NaN       9.303359       4.041882   
min         0.000000             NaN     -34.603410     -99.314630   
25%     15954.000000             NaN       4.225305      -1.108172   
50%     29089.000000             NaN       9.335275       0.381605   
75%     40921.750000             NaN      11.952784       2.166844   
max     65535.000000             NaN     103.960279      71.385898   

               Acc_Z      FreeAcc_X      FreeAcc_Y      FreeAcc_Z  \
count  558656.000000  558656.000000  558656.000000  558656.000000   
mean       -0.660292       0.025784       0.011230       0.007669   
std         5.832908       4.405676       6.119720       6.259061   
min      -107.526134    -105.815792     -79.174587     -49.338157   
25%        -2.505191      -1.151191      -1.779894      -2.389813   
50%        -0.519209       0.017844       0.000886       0.005823   
75%         1.766701       1.174968       1.782629       2.405411   
max        75.945099      96.742472      96.015080     697.739821   

               Gyr_X          Gyr_Y          Gyr_Z          Mag_X  \
count  558656.000000  558656.000000  558656.000000  558656.000000   
mean       -0.004178       0.011052       0.003737      -0.557325   
std         1.632140       2.214487       1.105100       0.636081   
min       -33.702383      -9.057740      -6.606975      -2.298096   
25%        -0.675111      -1.191530      -0.348207      -0.908936   
50%        -0.001527      -0.089044      -0.013546      -0.795654   
75%         0.690230       0.341896       0.231971      -0.618164   
max        31.032621      10.805331       7.220805       3.220947   

               Mag_Y          Mag_Z       VelInc_X       VelInc_Y  \
count  558656.000000  558656.000000  558656.000000  558656.000000   
mean       -0.042844      -0.017553       0.072629       0.005707   
std         0.503955       0.403815       0.093800       0.040689   
min        -1.884277      -4.077148      -0.393167      -1.019948   
25%        -0.474854      -0.297607       0.042065      -0.011118   
50%        -0.063965      -0.007568       0.093340       0.003787   
75%         0.437500       0.259766       0.119520       0.021781   
max         1.720703       3.075439       6.843565       0.691373   

            VelInc_Z      OriInc_q0      OriInc_q1      OriInc_q2  \
count  558656.000000  558656.000000  558656.000000  558656.000000   
mean       -0.006537       0.999890      -0.000022       0.000055   
std         0.058681       0.000234       0.008171       0.011078   
min        -1.774539       0.985459      -0.167711      -0.092002   
25%        -0.025152       0.999872      -0.003377      -0.005960   
50%        -0.004962       0.999971      -0.000008      -0.000445   
75%         0.017876       0.999995       0.003453       0.001711   
max         0.749798       1.000000       0.154538       0.080873   

           OriInc_q3           Roll          Pitch            Yaw  Latitude  \
count  558656.000000  558656.000000  558656.000000  558656.000000       0.0   
mean        0.000019      -4.792646     -47.203698       5.860481       NaN   
std         0.005527     117.782133      53.845210      97.297090       NaN   
min        -0.047890    -179.999984     -89.980865    -179.999875       NaN   
25%        -0.001742    -138.420069     -78.673051     -76.479373       NaN   
50%        -0.000068       3.496418     -71.074598     -10.139154       NaN   
75%         0.001160      66.495412     -51.988967      98.717686       NaN   
max         0.040124     179.999941      89.896154     179.999286       NaN   

       Longitude  Altitude  Vel_X  Vel_Y  Vel_Z  
count        0.0  558714.0    0.0    0.0    0.0  
mean         NaN       0.0    NaN    NaN    NaN  
std          NaN       0.0

In [5]:
# Drop NaN columns
data.dropna(axis=1, how="all", inplace=True)

In [6]:
# Reinspect the data

data.describe()

PacketCounter          Acc_X          Acc_Y          Acc_Z  \
count  558714.000000  558656.000000  558656.000000  558656.000000   
mean    29984.349316       7.265798       0.567765      -0.660292   
std     17730.810665       9.303359       4.041882       5.832908   
min         0.000000     -34.603410     -99.314630    -107.526134   
25%     15954.000000       4.225305      -1.108172      -2.505191   
50%     29089.000000       9.335275       0.381605      -0.519209   
75%     40921.750000      11.952784       2.166844       1.766701   
max     65535.000000     103.960279      71.385898      75.945099   

           FreeAcc_X      FreeAcc_Y      FreeAcc_Z          Gyr_X  \
count  558656.000000  558656.000000  558656.000000  558656.000000   
mean        0.025784       0.011230       0.007669      -0.004178   
std         4.405676       6.119720       6.259061       1.632140   
min      -105.815792     -79.174587     -49.338157     -33.702383   
25%        -1.151191      -1.779894      -2.389813      -0.675111   
50%         0.017844       0.000886       0.005823      -0.001527   
75%         1.174968       1.782629       2.405411       0.690230   
max        96.742472      96.015080     697.739821      31.032621   

               Gyr_Y          Gyr_Z          Mag_X          Mag_Y  \
count  558656.000000  558656.000000  558656.000000  558656.000000   
mean        0.011052       0.003737      -0.557325      -0.042844   
std         2.214487       1.105100       0.636081       0.503955   
min        -9.057740      -6.606975      -2.298096      -1.884277   
25%        -1.191530      -0.348207      -0.908936      -0.474854   
50%        -0.089044      -0.013546      -0.795654      -0.063965   
75%         0.341896       0.231971      -0.618164       0.437500   
max        10.805331       7.220805       3.220947       1.720703   

               Mag_Z       VelInc_X       VelInc_Y       VelInc_Z  \
count  558656.000000  558656.000000  558656.000000  558656.000000   
mean       -0.017553       0.072629       0.005707      -0.006537   
std         0.403815       0.093800       0.040689       0.058681   
min        -4.077148      -0.393167      -1.019948      -1.774539   
25%        -0.297607       0.042065      -0.011118      -0.025152   
50%        -0.007568       0.093340       0.003787      -0.004962   
75%         0.259766       0.119520       0.021781       0.017876   
max         3.075439       6.843565       0.691373       0.749798   

           OriInc_q0      OriInc_q1      OriInc_q2      OriInc_q3  \
count  558656.000000  558656.000000  558656.000000  558656.000000   
mean        0.999890      -0.000022       0.000055       0.000019   
std         0.000234       0.008171       0.011078       0.005527   
min         0.985459      -0.167711      -0.092002      -0.047890   
25%         0.999872      -0.003377      -0.005960      -0.001742   
50%         0.999971      -0.000008      -0.000445      -0.000068   
75%         0.999995       0.003453       0.001711       0.001160   
max         1.000000       0.154538       0.080873       0.040124   

                Roll          Pitch            Yaw  Altitude  
count  558656.000000  558656.000000  558656.000000  558714.0  
mean       -4.792646     -47.203698       5.860481       0.0  
std       117.782133      53.845210      97.297090       0.0  
min      -179.999984     -89.980865    -179.999875       0.0  
25%      -138.420069     -78.673051     -76.479373       0.0  
50%         3.496418     -71.074598     -10.139154       0.0  
75%        66.495412     -51.988967      98.717686       0.0  
max       179.999941      89.896154     179.999286       0.0

**Data ranges are quite wide in some columns, e.g. Acc_X (min: -34, median: 9.33, max: 103.96), meaning that we should scale to improve model performance**

In [7]:
data.sample(5)

Surface SensorLocation  PacketCounter      Acc_X     Acc_Y      Acc_Z  \
362400   CALIB     Left shank          26213  10.160310  0.995376   1.015983   
30323     SlpD     Left shank          17905   1.363751 -8.687628 -25.386734   
388447   CALIB     Left thigh          26699   9.902317  0.029394  -2.751229   
136409    StrD    Right shank          44337  -7.333302 -1.536548  -5.682857   
518335    SlpU    Right thigh          64837   8.070260 -2.161805  -0.811349   

        FreeAcc_X  FreeAcc_Y  FreeAcc_Z     Gyr_X     Gyr_Y     Gyr_Z  \
362400  -1.252149  -0.001744   0.369983  0.332693 -0.405197 -0.275952   
30323    6.941194 -23.112673 -21.620099  0.795797 -2.100600  0.347143   
388447   0.307643  -1.011377   0.410250  0.055633  0.125394  0.059994   
136409   0.865468  -9.297049  -8.698093  2.362065  0.100068 -1.981625   
518335  -2.743735  -0.869340  -1.927775 -0.604639 -2.883929  0.220496   

           Mag_X     Mag_Y     Mag_Z  VelInc_X  VelInc_Y  VelInc_Z  OriInc_q0  \
362400 -0.840332  0.214355 -0.047119  0.101596  0.009796  0.010382   0.999996   
30323  -1.709717  1.099854 -0.415283  0.016455 -0.085839 -0.254047   0.999935   
388447 -0.715332  0.474609  0.065430  0.099006  0.000331 -0.027574   1.000000   
136409 -0.337646 -0.279053  1.031982 -0.073505 -0.013965 -0.056962   0.999881   
518335 -1.128662 -0.831299 -0.375977  0.080832 -0.021551 -0.006883   0.999891   

        OriInc_q1  OriInc_q2  OriInc_q3        Roll      Pitch         Yaw  \
362400   0.001663  -0.002026  -0.001380   62.612347 -75.794295 -160.448520   
30323    0.003979  -0.010503   0.001736   16.167717 -61.581189  -70.005503   
388447   0.000278   0.000627   0.000300 -175.807346 -80.326214   94.586319   
136409   0.011810   0.000500  -0.009908  138.570091 -15.183699  127.745609   
518335  -0.003023  -0.014419   0.001102  -13.946699 -80.628377 -108.390986   

        Altitude  
362400       0.0  
30323        0.0  
388447       0.0  
136409       0.0  
518335       0.0

In [8]:
# Some columns are just meaningless
print(data[["PacketCounter"]].join(data[["Altitude"]]))
data.drop(["PacketCounter", "Altitude"], axis=1, inplace=True)

        PacketCounter  Altitude
0               63872       0.0
1               63873       0.0
2               63874       0.0
3               63875       0.0
4               63876       0.0
...               ...       ...
558709          25830       0.0
558710          25831       0.0
558711          25832       0.0
558712          25833       0.0
558713          25834       0.0

[558714 rows x 2 columns]


#### Bring all columns to numeric data formats

In [13]:
data["Surface"], _ = data["Surface"].factorize()
data["SensorLocation"], _ = data["SensorLocation"].factorize()

In [14]:
data.dtypes

Surface             int64
SensorLocation      int64
Acc_X             float64
Acc_Y             float64
Acc_Z             float64
FreeAcc_X         float64
FreeAcc_Y         float64
FreeAcc_Z         float64
Gyr_X             float64
Gyr_Y             float64
Gyr_Z             float64
Mag_X             float64
Mag_Y             float64
Mag_Z             float64
VelInc_X          float64
VelInc_Y          float64
VelInc_Z          float64
OriInc_q0         float64
OriInc_q1         float64
OriInc_q2         float64
OriInc_q3         float64
Roll              float64
Pitch             float64
Yaw               float64
dtype: object

In [15]:
data

Surface  SensorLocation      Acc_X     Acc_Y     Acc_Z  FreeAcc_X  \
0             0               0   9.855517  1.419511  1.056100   0.000000   
1             0               0   9.618346  1.531807  0.793353   0.385820   
2             0               0   9.340269  1.537687  0.719518   0.436632   
3             0               0   9.262140  1.534052  0.871657   0.379882   
4             0               0   9.087550  2.048158  0.982283   0.825285   
...         ...             ...        ...       ...       ...        ...   
558709        5               0  12.380745  5.285336 -1.026498   1.841389   
558710        5               0  11.609036  2.817452 -1.885530   0.166782   
558711        5               0  10.621329  1.171285 -2.344570  -0.925919   
558712        5               0  10.188926  0.569020 -2.817934  -1.261935   
558713        5               0   9.914534  0.910060 -2.411476  -1.132452   

        FreeAcc_Y  FreeAcc_Z     Gyr_X     Gyr_Y     Gyr_Z     Mag_X  \
0       -0.000000   0.200383  0.016545 -0.029152 -0.009963 -0.963623   
1       -0.297160  -0.053012  0.061748 -0.028399 -0.009419 -0.958984   
2       -0.315490  -0.334679  0.084028 -0.025894 -0.019274 -0.956787   
3       -0.161510  -0.393031  0.130678 -0.017727 -0.027842 -0.956787   
4        0.149745  -0.483171  0.119649 -0.008019 -0.037221 -0.954346   
...           ...        ...       ...       ...       ...       ...   
558709   0.373530   3.556620 -0.113998 -1.496558  0.001963 -0.900635   
558710  -1.240775   2.216199 -1.263676 -1.431789  0.011664 -0.907715   
558711  -2.178039   0.868089 -1.487070 -1.169878 -0.289941 -0.909668   
558712  -2.763749   0.328654 -0.591665 -0.954093 -0.473706 -0.923828   
558713  -2.190064   0.130235  0.530131 -0.762595 -0.713394 -0.916504   

           Mag_Y     Mag_Z  VelInc_X  VelInc_Y  VelInc_Z  OriInc_q0  \
0       0.235596 -0.266357  0.098554  0.014189  0.010577   1.000000   
1       0.235596 -0.266357  0.096183  0.015311  0.007952   1.000000   
2       0.237793 -0.266357  0.093403  0.015365  0.007214   1.000000   
3       0.244385 -0.268799  0.092623  0.015322  0.008735   1.000000   
4       0.237793 -0.266357  0.090879  0.020459  0.009839   1.000000   
...          ...       ...       ...       ...       ...        ...   
558709  0.083740 -0.221680  0.123879  0.052849 -0.009368   0.999972   
558710  0.096680 -0.204834  0.116221  0.028065 -0.018201   0.999954   
558711  0.074951 -0.197510  0.106365  0.011387 -0.022909   0.999954   
558712  0.087646 -0.184814  0.102035  0.005366 -0.027709   0.999981   
558713  0.074707 -0.182129  0.099268  0.008810 -0.023713   0.999983   

        OriInc_q1  OriInc_q2  OriInc_q3        Roll      Pitch         Yaw  
0        0.000083  -0.000146  -0.000050   53.299628 -79.819856 -120.515449  
1        0.000309  -0.000142  -0.000047   46.646334 -79.631895 -115.706065  
2        0.000420  -0.000129  -0.000096   46.794904 -79.634026 -115.808157  
3        0.000653  -0.000089  -0.000139   46.968656 -79.627834 -115.908051  
4        0.000598  -0.000040  -0.000186   47.134912 -79.615354 -116.007378  
...           ...        ...        ...         ...        ...         ...  
558709  -0.000570  -0.007483   0.000010  101.118168 -74.405827 -166.211231  
558710  -0.006318  -0.007159   0.000058  103.259561 -74.237416 -169.185534  
558711  -0.007435  -0.005849  -0.001450  104.609544 -73.906827 -171.468586  
558712  -0.002958  -0.004770  -0.002369  105.833962 -73.499663 -173.095611  
558713   0.002651  -0.003813  -0.003567  107.202890 -72.975216 -174.200949  

[558714 rows x 24 columns]

In [22]:
# There are still some NaN rows, which we'll get rid off
na_rows = data[data.Acc_X.isna()]
print("Total NaN rows: ", len(na_rows))
print("Samples of NaN rows:")
na_rows[:5]

Total NaN rows:  58
Samples of NaN rows:


Surface  SensorLocation  Acc_X  Acc_Y  Acc_Z  FreeAcc_X  FreeAcc_Y  \
13557        5               5    NaN    NaN    NaN        NaN        NaN   
13558        5               5    NaN    NaN    NaN        NaN        NaN   
13559        5               5    NaN    NaN    NaN        NaN        NaN   
13560        5               5    NaN    NaN    NaN        NaN        NaN   
15849        5               3    NaN    NaN    NaN        NaN        NaN   

       FreeAcc_Z  Gyr_X  Gyr_Y  Gyr_Z  Mag_X  Mag_Y  Mag_Z  VelInc_X  \
13557        NaN    NaN    NaN    NaN    NaN    NaN    NaN       NaN   
13558        NaN    NaN    NaN    NaN    NaN    NaN    NaN       NaN   
13559        NaN    NaN    NaN    NaN    NaN    NaN    NaN       NaN   
13560        NaN    NaN    NaN    NaN    NaN    NaN    NaN       NaN   
15849        NaN    NaN    NaN    NaN    NaN    NaN    NaN       NaN   

       VelInc_Y  VelInc_Z  OriInc_q0  OriInc_q1  OriInc_q2  OriInc_q3  Roll  \
13557       NaN       NaN        NaN        NaN        NaN        NaN   NaN   
13558       NaN       NaN        NaN        NaN        NaN        NaN   NaN   
13559       NaN       NaN        NaN        NaN        NaN        NaN   NaN   
13560       NaN       NaN        NaN        NaN        NaN        NaN   NaN   
15849       NaN       NaN        NaN        NaN        NaN        NaN   NaN   

       Pitch  Yaw  
13557    NaN  NaN  
13558    NaN  NaN  
13559    NaN  NaN  
13560    NaN  NaN  
15849    NaN  NaN

In [23]:
data.dropna(axis=0, inplace=True)

In [24]:
# Count rows after dropping NaN rows
data.shape

(558656, 24)

#### Before scaling, let's apply Butterworth filter

In [25]:
from scipy import signal

sos = signal.butter(2, 6, btype="lp", fs=100, output="sos")
for c in data.columns[2:]:
    data[c] = signal.sosfilt(sos, data[c])

In [29]:
data

Surface  SensorLocation      Acc_X     Acc_Y     Acc_Z  FreeAcc_X  \
0             0               0   0.274572  0.039547  0.029423   0.000000   
1             0               0   1.222236  0.180121  0.124361   0.010749   
2             0               0   2.712953  0.410293  0.259897   0.049522   
3             0               0   4.331996  0.670756  0.396961   0.112421   
4             0               0   5.828968  0.934256  0.529149   0.193133   
...         ...             ...        ...       ...       ...        ...   
558709        5               0  10.964821 -0.021870 -4.546547  -0.945116   
558710        5               0  11.195818  0.945932 -3.477958  -0.702196   
558711        5               0  11.371426  1.745407 -2.662180  -0.446572   
558712        5               0  11.406374  2.179738 -2.148389  -0.328878   
558713        5               0  11.295616  2.281439 -1.896937  -0.350812   

        FreeAcc_Y  FreeAcc_Z     Gyr_X     Gyr_Y     Gyr_Z     Mag_X  \
0        0.000000   0.005583  0.000461 -0.000812 -0.000278 -0.026846   
1       -0.008279   0.017925  0.003322 -0.003614 -0.001227 -0.120021   
2       -0.037562   0.016477  0.010874 -0.007971 -0.002987 -0.268268   
3       -0.080921  -0.017284  0.024137 -0.012369 -0.005799 -0.432067   
4       -0.110968  -0.079858  0.042188 -0.015503 -0.009929 -0.586609   
...           ...        ...       ...       ...       ...       ...   
558709  -5.098233   0.546932  0.743814 -2.219710 -0.728531 -0.894850   
558710  -3.628974   1.128230  1.284100 -1.859240 -0.690389 -0.900004   
558711  -2.481636   1.590433  1.343089 -1.594546 -0.598439 -0.903746   
558712  -1.764614   1.803741  1.093489 -1.393152 -0.506810 -0.906942   
558713  -1.422822   1.774054  0.765510 -1.226699 -0.450900 -0.910101   

           Mag_Y     Mag_Z  VelInc_X  VelInc_Y  VelInc_Z  OriInc_q0  \
0       0.006564 -0.007421  0.002746  0.000395  0.000295   0.027860   
1       0.029375 -0.033211  0.012222  0.001800  0.001246   0.124686   
2       0.065806 -0.074329  0.027129  0.004101  0.002604   0.279059   
3       0.106477 -0.119930  0.043320  0.006704  0.003978   0.450005   
4       0.145348 -0.163148  0.058290  0.009336  0.005302   0.611627   
...          ...       ...       ...       ...       ...        ...   
558709  0.119405 -0.260375  0.110250 -0.001151 -0.043889   0.999608   
558710  0.110911 -0.244720  0.112378  0.008737 -0.033410   0.999704   
558711  0.103374 -0.231352  0.113994  0.016921 -0.025449   0.999789   
558712  0.096742 -0.219585  0.114246  0.021404 -0.020497   0.999858   
558713  0.091121 -0.209083  0.113077  0.022511 -0.018149   0.999911   

        OriInc_q1  OriInc_q2  OriInc_q3        Roll      Pitch         Yaw  
0        0.000002  -0.000004  -0.000001    1.484915  -2.223763   -3.357532  
1        0.000017  -0.000018  -0.000006    6.460350  -9.947169  -14.892581  
2        0.000054  -0.000040  -0.000015   14.048316 -22.251084  -33.034120  
3        0.000121  -0.000062  -0.000029   22.151792 -35.866961  -52.905943  
4        0.000211  -0.000078  -0.000050   29.673240 -48.734880  -71.589983  
...           ...        ...        ...         ...        ...         ...  
558709   0.003721  -0.011096  -0.003643   84.097977 -76.051035 -153.345997  
558710   0.006421  -0.009295  -0.003452   89.171513 -75.808127 -157.842392  
558711   0.006716  -0.007972  -0.002992   93.697180 -75.486002 -161.726226  
558712   0.005467  -0.006965  -0.002534   97.565948 -75.118826 -165.073126  
558713   0.003827  -0.006133  -0.002254  100.762009 -74.719617 -167.916971  

[558656 rows x 24 columns]

In [26]:
# Reinspect data
data.describe()

Surface  SensorLocation          Acc_X          Acc_Y  \
count  558656.000000   558656.000000  558656.000000  558656.000000   
mean        4.348073        2.499876       7.265727       0.567752   
std         2.905121        1.707821       8.614531       2.550242   
min         0.000000        0.000000     -34.242811     -16.430477   
25%         2.000000        1.000000       5.254875      -0.897328   
50%         5.000000        2.000000       9.477075       0.344384   
75%         7.000000        4.000000      11.933736       1.965163   
max         9.000000        5.000000      33.277815      21.788693   

               Acc_Z      FreeAcc_X      FreeAcc_Y      FreeAcc_Z  \
count  558656.000000  558656.000000  558656.000000  558656.000000   
mean       -0.660281       0.025788       0.011237       0.007660   
std         4.420545       3.274880       5.077369       4.497234   
min       -31.334842     -29.697894     -30.389459     -26.869339   
25%        -2.476811      -0.939218      -1.669901      -1.835062   
50%        -0.583415       0.024018       0.002227       0.026317   
75%         1.503081       0.994179       1.651147       2.333952   
max        23.332535      28.161930      31.168744     121.969546   

               Gyr_X          Gyr_Y          Gyr_Z          Mag_X  \
count  558656.000000  558656.000000  558656.000000  558656.000000   
mean       -0.004179       0.011058       0.003740      -0.557319   
std         1.059509       2.153199       1.069880       0.635438   
min        -8.140505      -7.484386      -4.530060      -2.292965   
25%        -0.567253      -1.151508      -0.326936      -0.908822   
50%        -0.006245      -0.096573      -0.014353      -0.795440   
75%         0.590471       0.316637       0.222279      -0.617712   
max        11.003526      10.484148       6.992601       3.085087   

               Mag_Y          Mag_Z       VelInc_X       VelInc_Y  \
count  558656.000000  558656.000000  558656.000000  558656.000000   
mean       -0.042845      -0.017552       0.072628       0.005707   
std         0.503457       0.403099       0.086213       0.025894   
min        -1.877459      -3.943682      -0.348027      -0.159699   
25%        -0.474406      -0.297045       0.052416      -0.009007   
50%        -0.064668      -0.007610       0.094780       0.003430   
75%         0.436947       0.259388       0.119463       0.019719   
max         1.718803       1.978990       1.276434       0.224424   

            VelInc_Z      OriInc_q0      OriInc_q1      OriInc_q2  \
count  558656.000000  558656.000000  558656.000000  558656.000000   
mean       -0.006537       0.999883      -0.000022       0.000055   
std         0.044602       0.002238       0.005305       0.010768   
min        -0.320558       0.027860      -0.040689      -0.037410   
25%        -0.024787       0.999859      -0.002839      -0.005763   
50%        -0.005542       0.999966      -0.000031      -0.000483   
75%         0.015295       0.999994       0.002954       0.001585   
max         0.230868       1.045715       0.054988       0.052397   

           OriInc_q3           Roll          Pitch            Yaw  
count  558656.000000  558656.000000  558656.000000  558656.000000  
mean        0.000019      -4.793356     -47.203209       5.861633  
std         0.005350     112.390474      53.788027      93.967385  
min        -0.025530    -196.286511     -94.688277    -196.132527  
25%        -0.001636    -121.563393     -78.613913     -75.258293  
50%        -0.000072       4.092174     -71.090969      -6.093169  
75%         0.001113      62.945688     -51.916508      97.013466  
max         0.034952     196.788804      89.633476     196.793206

**Value ranges are a bit narrower now**

Dump filtered dataset into a CSV file for further analysis

In [ ]:
data.to_csv("filtered_dataset_sample.csv")